File: RQ3/C-REval-Desc/task74_desc.c
#include <stdio.h>
#include <stdlib.h>

int digits_sum(int n) {
    int neg = 1;
    if (n < 0) {
        n = -n;
        neg = -1;
    }
    int sum = 0;
    while (n > 0) {
        sum += n % 10;
        n /= 10;
    }
    return sum * neg;
}

int compare(const void *a, const void *b) {
    int sum_a = digits_sum(*(int *)a);
    int sum_b = digits_sum(*(int *)b);
    return (sum_a > sum_b) - (sum_a < sum_b);
}

void order_by_points(int *nums, int size) {
    qsort(nums, size, sizeof(int), compare);
}

int main() {
    int nums[] = {1, 11, -1, -11, -12};
    int size = sizeof(nums) / sizeof(nums[0]);
    
    order_by_points(nums, size);
    
    for (int i = 0; i < size; i++) {
        printf("%d ", nums[i]);
    }
    
    return 0;
}

File: RQ3/Java-REval-Desc/task74_desc.java
public class Solution {
    private static int digits_sum(int n) {
        int neg = 1;
        if (n < 0) {
            n = -1 * n;
            neg = -1;
        }
        int sum = 0;
        for (char digit : String.valueOf(n).toCharArray()) {
            sum += Character.getNumericValue(digit);
        }
        return sum * neg;
    }

    public static void order_by_points(int[] nums) {
        int[] _result = Arrays.stream(nums)
            .boxed()
            .sorted(Comparator.comparingInt(Solution::digits_sum))
            .mapToInt(Integer::intValue)
            .toArray();

        assert func(_result);  // POST: func is equivalent to: Write a function which sorts the given list of integers in ascending order according to the sum of their digits. Note: if there are several items with similar sum of their digits, order them based on their index in original list. For example: >>> order_by_points([1, 11, -1, -11, -12]) == [-1, -11, 1, -12, 11] >>> order_by_points([]) == []
    }
}
